tests/test_reader.py

```diff
@@ -293,15 +293,12 @@ def test_empty_shards_index_json_exception(local_remote_dir: Tuple[str, str]):
 
 
 @pytest.mark.usefixtures('mds_dataset_dir')
-def test_accidental_shard_delete_exception(mds_dataset_dir: Any):
+def test_accidental_shard_delete(mds_dataset_dir: Any):
     remote_dir, local_dir = mds_dataset_dir
-    filename = 'shard.00000.mds'
+    basename = 'shard.00000.mds'
+    filename = os.path.join(local_dir, basename)
     dataset = StreamingDataset(local=local_dir, remote=remote_dir)
-
-    with pytest.raises(RuntimeError,
-                       match=f'.*Check if the shard file exists in your remote location.*'):
-        for _ in dataset:
-            if os.path.exists(os.path.join(local_dir, filename)):
-                os.remove(os.path.join(local_dir, filename))
-            pass
+    for _ in dataset:
+        if os.path.exists(filename):
+            os.remove(filename)
```

Comment:
Can you validate that the `filename` is present at the end of the test ?

Reply:
Done in https://github.com/mosaicml/streaming/pull/330/commits/380f9627e4fcb8b7b9a1767bc0ca479d1bd2bcab.